packages/api/src/urlAdapter.ts

```diff
@@ -1,13 +1,13 @@
-import {PublicArticle} from './db/article'
-import {PublicPage} from './db/page'
+import {PublicArticle, PublicArticleUrlArgs} from './db/article'
+import {PublicPage, PublicPageUrlArgs} from './db/page'
 import {Author} from './db/author'
 import {PublicComment} from './db/comment'
 import {Peer} from './db/peer'
 
 export interface URLAdapter {
-  getPublicArticleURL(article: PublicArticle): string
+  getPublicArticleURL(article: PublicArticleUrlArgs): string
```

Comment:
You can't change the parameters. They are already in use and would break quite a few things.

But when calling `getPublicArticleURL()` you can create a "publicArticle" with the data you have and pass it like this. `getPublicArticleURL(fakeDraftArticle as PublicArticle)`.

Reply:
@roosnic1  done